app/controllers/notifications_controller.rb

```diff
@@ -21,9 +21,11 @@ def index
                      elsif params[:org_id].blank? && params[:filter].present?
                        filtered_notifications
                      else
-                       Notification.where(user_id: @user.id).order("notified_at DESC")
+                       @user.notifications
                      end
 
+    @notifications = @notifications.without_recently_aggregated_reactions_and_follows.order(notified_at: :desc)
```

Comment:
after selecting the correct set of notifications we filter out recently aggregated ones *before* ordering and pagination